search/fibonacci_search.cpp

```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+#include <vector>
+
+/*
+Input: a sorted array and a value
+Output: if the array contains the value, returns its index (index of its first occurence)
+        else returns -1
+*/
+
+int min(int first, int second){
+    if(first < second){
+        return first;
+    } else {
+        return second;
+    }
+}
+
+int fibonacci_search(std::vector<int> arr, int value){
```

Comment:
if will be nice if you pass the reference of  ``` vector<int> arr``` instead of creating another local copy, this will make program more efficient.

as this is a function for searching an element in the given vector it is understood that there is no need to modify this vector in this function so it will be more good if you pass the const reference of vector into this function.

Reply:
> if will be nice if you pass the reference of ` vector<int> arr` instead of creating another local copy, this will make program more efficient.
> 
> as this is a function for searching an element in the given vector it is understood that there is no need to modify this vector in this function so it will be more good if you pass the const reference of vector into this function.

I think what @ayaankhan98  is suggesting is this:

```suggestion
int fibonacci_search(const std::vector<int> &arr, int value){
```
this is now a pass-by-reference of a constant class that does not get modified. This is more efficient

Reply:
Thanks for reviewing. I changed the implementation from vector to array. I also added documentation and tests. Please inform me if there's anything to change ^^

Reply:
@sprintyaf c++ vectors are more stable than C-Style arrays use ```std::vector``` or if you wish to use arrays then try to use ```std::array``` instead of using c-style arrays